datafusion/substrait/tests/utils.rs

```diff
@@ -0,0 +1,171 @@
+    use std::fs::File;
+    use std::io::BufReader;
+    use std::sync::Arc;
+    use substrait::proto::read_rel::{NamedTable, ReadType};
+    use substrait::proto::rel::RelType;
+    use substrait::proto::{Plan, ReadRel, Rel};
+
+    pub fn read_json(path: &str) -> Plan {
+        serde_json::from_reader::<_, Plan>(BufReader::new(
+            File::open(path).expect("file not found"),
+        ))
+        .expect("failed to parse json")
+    }
+
+    pub struct TestSchemaCollector {
+        ctx: SessionContext,
+    }
+
+    impl TestSchemaCollector {
+        fn new() -> Self {
+            TestSchemaCollector {
+                ctx: SessionContext::new(),
+            }
+        }
+
+        pub fn generate_context_from_plan(plan: &Plan) -> SessionContext {
+            let mut schema_collector = Self::new();
+
+            for plan_rel in plan.relations.iter() {
+                match plan_rel.rel_type.as_ref().expect("blah") {
+                    substrait::proto::plan_rel::RelType::Rel(r) => {
+                        schema_collector.collect_schemas(r)
+                    }
+                    substrait::proto::plan_rel::RelType::Root(r) => schema_collector
+                        .collect_schemas(
+                            r.input.as_ref().expect("RelRoot must set input"),
+                        ),
+                }
+            }
+            schema_collector.ctx
+        }
+
+        fn collect_named_table(&mut self, read: &ReadRel, nt: &NamedTable) {
+            let table_reference = match nt.names.len() {
+                0 => {
+                    panic!("No table name found in NamedTable");
+                }
+                1 => TableReference::Bare {
+                    table: nt.names[0].clone().into(),
+                },
+                2 => TableReference::Partial {
+                    schema: nt.names[0].clone().into(),
+                    table: nt.names[1].clone().into(),
+                },
+                _ => TableReference::Full {
+                    catalog: nt.names[0].clone().into(),
+                    schema: nt.names[1].clone().into(),
+                    table: nt.names[2].clone().into(),
+                },
+            };
+
+            let substrait_schema = read
+                .base_schema
+                .as_ref()
+                .expect("No base schema found for NamedTable");
+            let empty_extensions = Extensions {
+                functions: Default::default(),
+                types: Default::default(),
+                type_variations: Default::default(),
+            };
+
+            let df_schema = from_substrait_named_struct(
+                substrait_schema,
+                &empty_extensions,
+                Some(table_reference.clone()),
+            )
+            .expect("Unable to generate DataFusion schema from Substrait NamedStruct");
+            let table = EmptyTable::new(df_schema.inner().clone());
+            self.ctx
```

Comment:
might be nicer to just return the name and schema (that'd also better match the name "collect") rather than modify ctx directly

Reply:
Good idea. Done.